src/components/autocomplete/AutocompleteConnected.tsx

```diff
@@ -0,0 +1,173 @@
+    const mod = (x: number, n: number) => (x % n + n) % n; // mod is a modulo function that works with negative numbers
+    const visibleItems = _.map(itemsWithHidden, (item: IItemBoxProps): IItemBoxProps => {
+        let active = false;
+        if (!item.hidden && !item.disabled) {
+            active = mod(activeIndex, visibleLength) === index;
+            index++;
+        }
+        return {...item, active};
+    });
+
+    return {
+        visibleItems,
+        active: autocomplete && autocomplete.active,
+        isOpen: autocomplete && autocomplete.open,
+        value: autocomplete && autocomplete.value || defaultValue,
+    };
+};
+
+const mapDispatchToProps = (dispatch: IDispatch, ownProps: IAutocompleteOwnProps): IAutocompleteDispatchProps => ({
+    onRender: () => dispatch(addAutocomplete(ownProps.id)),
+    onDestroy: () => dispatch(removeAutocomplete(ownProps.id)),
+    onDocumentClick: () => dispatch(toggleAutocomplete(ownProps.id, false)),
+    onFocus: () => dispatch(toggleAutocomplete(ownProps.id)),
+    onChange: (value: string) => dispatch(setAutocompleteValue(ownProps.id, value)),
+    setActive: (diff: number) => dispatch(setAutocompleteActive(ownProps.id, diff)),
+});
+
+@ReduxConnect(mapStateToProps, mapDispatchToProps)
+export class AutocompleteConnected extends React.Component<IAutocompleteProps, {}> {
+    private dropdown: HTMLDivElement;
+    private menu: HTMLDivElement;
+
+    componentWillMount() {
+        this.props.onRender();
+        document.addEventListener('mousedown', this.handleDocumentClick);
+    }
+
+    componentWillUnmount() {
+        document.removeEventListener('mousedown', this.handleDocumentClick);
+        this.props.onDestroy();
+    }
+
+    render() {
+        const pickerClasses = classNames('autocomplete', {open: this.props.isOpen});
+        const dropdownClasses = classNames('autocomplete-list-container bg-pure-white', {hidden: !this.props.isOpen});
+        return (
+            <div className={pickerClasses} ref={(ref: HTMLDivElement) => this.dropdown = ref}>
+                <div className='input-wrapper validate input-field'>
+                    <input
+                        onFocus={() => this.props.onFocus()}
+                        onKeyUp={(e: React.KeyboardEvent<HTMLInputElement>) => this.onKeyUp(e)}
+                        onKeyDown={(e: React.KeyboardEvent<HTMLInputElement>) => this.onKeyDown(e)}
+                        onChange={(e: React.ChangeEvent<HTMLInputElement>) => this.props.onChange(e.target.value)}
+                        value={this.props.value}
+                        required
+                    />
+                    {this.props.children}
+                </div>
+                <div className={dropdownClasses} ref={(ref: HTMLDivElement) => this.menu = ref}>
+                    <ListBoxConnected id={this.props.id} classes={['relative']} items={this.props.visibleItems} highlight={this.props.value} />

```

Comment:
Is the autocomplete behaviour will always be to push the content down? Would it be worth it to add the option for it to be floating like the dropdown?

Reply:
I agree, and I think the default behaviour should be absolutly positionned.